pytensor/link/numba/dispatch/linalg/solve/tridiagonal.py

```diff
@@ -0,0 +1,292 @@
+    [ndarray, ndarray, ndarray, ndarray, ndarray, float, str], tuple[ndarray, int]
+]:
+    ensure_lapack()
+    _check_scipy_linalg_matrix(dl, "gtcon")
+    _check_scipy_linalg_matrix(d, "gtcon")
+    _check_scipy_linalg_matrix(du, "gtcon")
+    _check_scipy_linalg_matrix(du2, "gtcon")
+    dtype = d.dtype
+    w_type = _get_underlying_float(dtype)
+    numba_gtcon = _LAPACK().numba_xgtcon(dtype)
+
+    def impl(
+        dl: ndarray,
+        d: ndarray,
+        du: ndarray,
+        du2: ndarray,
+        ipiv: ndarray,
+        anorm: float,
+        norm: str,
+    ) -> tuple[ndarray, int]:
+        n = np.int32(d.shape[-1])
+        rcond = np.empty(1, dtype=dtype)
+        work = np.empty(2 * n, dtype=dtype)
+        iwork = np.empty(n, dtype=np.int32)
+        info = val_to_int_ptr(0)
+
+        numba_gtcon(
+            val_to_int_ptr(ord(norm)),
+            val_to_int_ptr(n),
+            dl.view(w_type).ctypes,
+            d.view(w_type).ctypes,
+            du.view(w_type).ctypes,
+            du2.view(w_type).ctypes,
+            ipiv.ctypes,
+            np.array(anorm, dtype=dtype).view(w_type).ctypes,
+            rcond.view(w_type).ctypes,
+            work.view(w_type).ctypes,
+            iwork.ctypes,
+            info,
+        )
+
+        return rcond, int_ptr_to_val(info)
+
+    return impl
+
+
+def _solve_tridiagonal(
+    a: ndarray,
+    b: ndarray,
+    lower: bool,
+    overwrite_a: bool,
+    overwrite_b: bool,
+    check_finite: bool,
+    transposed: bool,
+):
+    """
+    Solve a positive-definite linear system using the Cholesky decomposition.
+    """
+    return linalg.solve(
+        a=a,
+        b=b,
+        lower=lower,
+        overwrite_a=overwrite_a,
+        overwrite_b=overwrite_b,
+        check_finite=check_finite,
+        transposed=transposed,
+        assume_a="tridiagonal",
+    )
+
+
+@overload(_solve_tridiagonal)
+def _tridiagonal_solve_impl(
+    A: ndarray,
+    B: ndarray,
+    lower: bool,
+    overwrite_a: bool,
+    overwrite_b: bool,
+    check_finite: bool,
+    transposed: bool,
+) -> Callable[[ndarray, ndarray, bool, bool, bool, bool, bool], ndarray]:
+    ensure_lapack()
+    _check_scipy_linalg_matrix(A, "solve")
+    _check_scipy_linalg_matrix(B, "solve")
+
+    def impl(
+        A: ndarray,
+        B: ndarray,
+        lower: bool,
+        overwrite_a: bool,
+        overwrite_b: bool,
+        check_finite: bool,
+        transposed: bool,
+    ) -> ndarray:
+        n = np.int32(A.shape[-1])
+        _solve_check_input_shapes(A, B)
+        norm = "1"
+
+        if transposed:
+            A = A.T
+        dl, d, du = np.diag(A, -1), np.diag(A, 0), np.diag(A, 1)
+
+        # Adapted from scipy _matrix_norm_tridiagonal:
+        # https://github.com/scipy/scipy/blob/0f1fd4a7268b813fa2b844ca6038e4dfdf90084a/scipy/linalg/_basic.py#L356-L367
+        anorm = np.abs(d)
+        anorm[1:] += np.abs(du)
+        anorm[:-1] += np.abs(dl)
+        anorm = anorm.max()
```

Comment:
nit: this should be split out to a separate function

Reply:
Sure but numba loves compiling nestes functions

Reply:
if there's a performance advantage to keeping it inlined, keep it inlined.

I was thinking if you break it out, it fits the template of the other solves with 3 functions: norm, decomp, and solve

Reply:
I don't love the norm/warnings at all, but that's a separate matter.

I can split, the only concern I could have is compile time but that's not why I inlined it. I'll do a quick benchmark to see if it's warranted

Reply:
Split it, if anything compile times seem a bit better, probably because at least that inner function can be cached.